crates/burn-tensor/src/tensor/activation/base.rs

```diff
@@ -78,6 +78,24 @@ pub fn softmax<const D: usize, B: Backend>(tensor: Tensor<B, D>, dim: usize) ->
     tensor.div(tensor_tmp)
 }
 
+/// Applies the softmin function on the input tensor along the given dimension.
+///
+/// `softmin(x_i) = exp(-x_i) / sum_j(exp(-x_j))`
+///
+/// # Notes
+///
+/// The dimension argument `dim` specifies the dimension along which the function will be computed.
+/// It must in the range of `0` and `D-1`.
+pub fn softmin<const D: usize, B: Backend>(tensor: Tensor<B, D>, dim: usize) -> Tensor<B, D> {
+    check!(TensorCheck::dim_ops::<D>("softmax", dim));
+
+    let tensor = tensor.clone() - tensor.detach().max_dim(dim);
+    let tensor = (-tensor).exp();
+    let tensor_tmp = tensor.clone().sum_dim(dim);
+
+    tensor.div(tensor_tmp)
```

Comment:
We could simplify this as `softmax(tensor.neg(), dim)` since softmin(x) is the equivalent of softmax(-x)